### PR TITLE
Fix AE integration (item voiding bug)

### PR DESCRIPTION
**Pull request by avlad171**
Latest build included a bug caused by some changes in SpecialInventoryHandler class. I overrode the getMultipleItems method to fix it as well as to improve code's efficiency (all the AE api functions called only once).

Also this fixes the empty ME network bug.

### CODE REVIEW

**Comment by bziemons** [comment]
Please make your changes on top of the current development branch and not your old branch. To solve this (and not open a new PR): rebase and force push or checkout and cherry-pick.

File: common/logisticspipes/proxy/specialinventoryhandler/AEInterfaceInventoryHandler.java [hunk 1]
```diff
@@ -62,7 +66,13 @@ public AEInterfaceInventoryHandler() {
 
 	@Override
 	public boolean isType(TileEntity tile, EnumFacing dir) {
-		return tile instanceof IGridHost && tile.hasCapability(LPStorageMonitorableAccessor.STORAGE_MONITORABLE_ACCESSOR_CAPABILITY, dir);
+		if (tile instanceof IGridHost && tile.hasCapability(LPStorageMonitorableAccessor.STORAGE_MONITORABLE_ACCESSOR_CAPABILITY, dir)) {
+			IGridNode tempNode = ((IGridHost) tile).getGridNode(AEPartLocation.fromFacing(dir));    //for some reason when AE loads (5 ticks) this is null
+			if (tempNode == null)																	//should not be a problem because interfaces are still treated like IInvertoryHandler
```
Comment (bziemons) [inline common/logisticspipes/proxy/specialinventoryhandler/AEInterfaceInventoryHandler.java]: That comment shouldn't fly away

File: common/logisticspipes/proxy/specialinventoryhandler/AEInterfaceInventoryHandler.java [hunk 2]
```diff
@@ -42,13 +44,15 @@
...
 	public IGridHost host;
+	public IGridNode node;
 	LinkedList<Entry<ItemIdentifier, Integer>> cached;
 
 	private AEInterfaceInventoryHandler(TileEntity tile, EnumFacing dir, boolean hideOnePerStack, boolean hideOne, int cropStart, int cropEnd) {
 		this.tile = tile;
 		this.hideOnePerStack = hideOnePerStack || hideOne;
 		this.acc = tile.getCapability(LPStorageMonitorableAccessor.STORAGE_MONITORABLE_ACCESSOR_CAPABILITY, dir);
-		host = (IGridHost) tile;
+		node = ((IGridHost)tile).getGridNode(AEPartLocation.fromFacing(dir));
```
Comment (bziemons) [inline common/logisticspipes/proxy/specialinventoryhandler/AEInterfaceInventoryHandler.java]: spacing, use the project's code style

File: common/logisticspipes/proxy/specialinventoryhandler/AEInterfaceInventoryHandler.java [hunk 3]
```diff
@@ -62,7 +66,13 @@ public AEInterfaceInventoryHandler() {
 
 	@Override
 	public boolean isType(TileEntity tile, EnumFacing dir) {
-		return tile instanceof IGridHost && tile.hasCapability(LPStorageMonitorableAccessor.STORAGE_MONITORABLE_ACCESSOR_CAPABILITY, dir);
+		if (tile instanceof IGridHost && tile.hasCapability(LPStorageMonitorableAccessor.STORAGE_MONITORABLE_ACCESSOR_CAPABILITY, dir)) {
+			IGridNode tempNode = ((IGridHost) tile).getGridNode(AEPartLocation.fromFacing(dir));    //for some reason when AE loads (5 ticks) this is null
```
Comment (bziemons) [inline common/logisticspipes/proxy/specialinventoryhandler/AEInterfaceInventoryHandler.java]: replace with `return ((IGridHost) tile).getGridNode(AEPartLocation.fromFacing(dir)) != null;`

File: common/logisticspipes/proxy/specialinventoryhandler/AEInterfaceInventoryHandler.java [hunk 4]
```diff
@@ -183,6 +212,10 @@ public int getSizeInventory() {
 		if (cached == null) {
 			initCache();
 		}
+
+		if(cached.size() == 0)
```
Comment (bziemons) [inline common/logisticspipes/proxy/specialinventoryhandler/AEInterfaceInventoryHandler.java]: spacing, code style

**Review by bziemons — CHANGES_REQUESTED** [review]

**Comment by bziemons** [comment]
If you can, please remove the merge commit by rebasing.

File: common/logisticspipes/proxy/specialinventoryhandler/AEInterfaceInventoryHandler.java (hunk 1 above)
Comment (avlad171) [reply common/logisticspipes/proxy/specialinventoryhandler/AEInterfaceInventoryHandler.java]: In idea and notepad++ looks perfectly fine. Anyway i'm going to remove it.

**Comment by bziemons** [comment]
If you haven't noticed: you added your own `output` folder. Anyway, I will merge this locally now. Please become familiar with git.

**Pull request closed without merging**